pkg/lint/lint.go

```diff
@@ -30,7 +30,7 @@ func All(basedir string, values map[string]interface{}, namespace string, strict
 
 	linter := support.Linter{ChartDir: chartDir}
 	rules.Chartfile(&linter)
-	rules.Values(&linter)
```

Comment:
Are there no concerns about changing the internals of this function per https://github.com/helm/helm/pull/7417#discussion_r375961183 where I tried to make a similar change? While this function's signature isn't changing, it's output certainly might in cases where it was passed a non-empty values map, no?

Reply:
(Don't get me wrong, I'm all for this change. I was just curious)

Reply:
The issue that was raised with your change (which I hadn't seen until yesterday) was that you changed the public signature of `Values()`. That is an API breaking change, and is not allowed until Helm 4. @bacongobbler was not blocking your PR based on an abstract notion of "sameness." He was blocking it based on our policy to not introduce breaking changes to the public API. It's true that we do want to make sure things continue to function similarly, but I think you may have gotten sidetracked by that. Fixing a bug is, of course, a proper behavior change. He was just making a sidelong comment that if no values were provided, it should still function the same.

FWIW, I don't have a preference over which our our PRs gets merged. So if you'd rather update yours, I'll review the changes. If I had noticed it before writing this one, I would have offered feedback then instead of doing what equates to duplicate work.

Reply:
Thanks for the explanation! I'm fine with this moving forward since I'd have to resolve conflicts and you already have approvals. Thanks for making the fix!

Reply:
Can you also mark that this closes https://github.com/helm/helm/issues/7273?

Reply:
@robbie-demuth that is a good questions. Notice that the `Values` function is still present with the same API definition. There is now a new public function named `ValuesWithOverrides` with the new signature. The Go public API change was additive in the form of a new function with the new signature.